### ISSUE TITLE
Cannot use Identity token as Password using JFrog CLI

### ISSUE THREAD

**Issue by SharonSbo** [jfrog/jfrog-cli#1616]
While trying to use identity token as password (i.e username + Identity Token) in the jf CLI, the response is 401.
The same command works fine when using API request.

**To Reproduce**
1. Login to Artifactory UI.
2. Create a new Identity Token through User Profile.
3. Copy the token and use it in the JFrog CLI configuration under password/API key.
4. Run the command `jf rt ping --server-id <SERVER_NAME>`
**The output:**
`10:46:10 [🚨Error] Server response: 401 Unauthorized
{
 "errors": [
  {
   "status": 401,
   "message": "Bad credentials"
  }
 ]
}`

**Expected behavior**
 The identity token will work as password using CLI.

**Screenshots**
![image](https://user-images.githubusercontent.com/109900451/181026008-92c21670-179a-4019-865a-638c037e44e5.png)
![image](https://user-images.githubusercontent.com/109900451/181026187-f26ca0f7-655a-43c8-a976-f5e22af0ba71.png)
![image](https://user-images.githubusercontent.com/109900451/181026295-1f16250f-bd27-4f7c-8ab6-85ff2659c7a2.png)

**Versions**
- JFrog CLI version: tested on 2.21.3 
- Artifactory Version: 7.41.6

**Comment by scott-hpe** [comment]
HPE is impacted by this issue.

**Comment by pem-imre** [comment]
This issue seems to affect the [npm integration commands](https://www.jfrog.com/confluence/display/CLI/CLI+for+JFrog+Artifactory#CLIforJFrogArtifactory-BuildingNpmPackagesUsingtheNpmClient). 

With jf v2.22.0 I get this output:
'''
$ jf rt npmi
13:56:50 [🟠Warn] You are using a deprecated syntax of the command.
        The new command syntax is quite similar to the syntax used by the native npm client.
        All you need to do is to add 'jf' as a prefix to the command.
        For example:
        $ jf npm install ...
        The --build-name and --build-number options are still supported.
13:56:50 [🔵Info] Running npm install.
13:57:00 [🚨Error] received invalid access-token
'''

**Comment by yahavi** [comment]
@SharonSbo @scott-hpe @pem-imre 
Thanks for reporting this issue!
We are currently working on supporting identity tokens as passwords.

**Workaround**
Add `--enc-password=false` & `--basic-auth-only` flags to the `jf c add` or `jf c edit` commands:
```
jf c add <server-id> --enc-password=false --basic-auth-only

jf c edit <server-id> --enc-password=false --basic-auth-only
```

**Root causes**
1. After `jf c add` we send a GET request to [/api/security/encryptedPassword](https://www.jfrog.com/confluence/display/JFROG/Artifactory+REST+API#ArtifactoryRESTAPI-GetUserEncryptedPassword). This request doesn't work with identity tokens. However, this rest API encrypts only raw text passwords and it is not needed for such tokens. Therefore it is safe to add the `--enc-password=false` flag as a workaround.
2. For improved security, the JFrog CLI uses an automatically-token-refreshment mechanism. AFAIK, creating a regular access token with an identity token is not supported. Therefore we suggest using the `--basic-auth-only` as a workaround.

We will keep you posted once a long-term solution is implemented.
We'll also appreciate any feedback about the workaround.

**Comment by yahavi** [comment]
@SharonSbo @scott-hpe @pem-imre 
This issue should be fixed in JFrog Artifactory 7.42.0. 
The version is already updated in the SaaS instances. If you're using an on-prem please feel free to upgrade your Artifactory server to the latest version.

We'd appreciate your feedback on that.
